guides/v3.5.0/tutorial/service.md

```diff
@@ -167,38 +123,30 @@ ember g component location-map
 
 Running this command generates three files: a component JavaScript file, a template, and a test file.
 
-Let's start by adding a `div` element to the component template.
-This `div` will act as a place for the 3rd party map API to render the map to.
-
-```handlebars {data-filename="app/templates/components/location-map.hbs" data-diff="-1,+2"}
-{{yield}}
-<div class="map-container"></div>
-```
-
-Next, update the component to append the map output to the `div` element we created.
-
-We provide the maps service into our component by initializing a property of our component, called `maps`.
+We provide the maps service into our component by initializing a property of our component, called `mapElement`.
 Services are commonly made available in components and other Ember objects by ["service injection"](../../applications/services/#toc_accessing-services).
 When you initialize a property with `import { inject } from '@ember/service';`,
 Ember tries to set that property with a service matching its name.
 
-With our `maps` service, our component will call the `getMapElement` function with the provided location.
+With our `mapElement` service, our component will call the `getMapElement` function with the provided location.
 We append the map element we get back from the service by implementing `didInsertElement`,
 which is a [component lifecycle hook](../../components/the-component-lifecycle/#toc_integrating-with-third-party-libraries-with-didinsertelement).
 This function runs during the component render, after the component's markup gets inserted into the page.
 
-```javascript {data-filename="app/components/location-map.js" data-diff="+2,+5,+7,+8,+9,+10,+11,+12"}
+```javascript {data-filename="app/components/location-map.js" data-diff="+2,+5,+6,+8,+9,+10,+11,+12"}
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
 
 export default Component.extend({
-  maps: service(),
+  classNames: ['map-container'],
+  mapElement: service(),
 
   didInsertElement() {
     this._super(...arguments);
-    let mapElement = this.maps.getMapElement(this.location);
-    let mapContainerElement = this.element.querySelector('.map-container');
-    mapContainerElement.appendChild(mapElement);
+    this.mapElement.getMapElement(this.location).then((mapElement) => {
+      this.$().append(mapElement);
```

Comment:
@sivakumar-kailasam made a good point that we should refer to this as `this.element` instead.